File: android/app/src/main/java/com/budgetmanager/domain/entity/Pocket.kt
package com.budgetmanager.domain.entity

import java.time.LocalDateTime

/**
 * Pocket represents a source of money (e.g., bank account, cash, e-wallet)
 */
data class Pocket(
    val id: Long = 0,
    val name: String,
    val description: String = "",
    val balance: Double = 0.0,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now()
)

File: server/internal/domain/pocket.go
package domain

import (
	"time"
)

// Pocket represents a source of money (e.g., bank account, cash, e-wallet)
type Pocket struct {
	ID          int64     `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description,omitempty"`
	Balance     float64   `json:"balance"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreatePocketRequest struct {
	Name        string  `json:"name"`
	Description string  `json:"description,omitempty"`
	Balance     float64 `json:"balance"`
}

type UpdatePocketRequest struct {
	Name        *string  `json:"name,omitempty"`
	Description *string  `json:"description,omitempty"`
	Balance     *float64 `json:"balance,omitempty"`
}
